libbpf-tools/tcppktlat.c

```diff
@@ -171,6 +230,134 @@ static void handle_lost_events(void *ctx, int cpu, __u64 lost_cnt)
 	fprintf(stderr, "lost %llu events on CPU #%d\n", lost_cnt, cpu);
 }
 
+static void calc_hist_stats(__u32 *slots, int slots_size, double *min,
+			    double *max, double *avg, double *mean, double *p95,
+			    double *p99)
+{
+	unsigned long long total = 0;
+	unsigned long long sum = 0;
+	int i, min_slot = -1, max_slot = -1;
+	unsigned long long low, high, mid;
+
+	/* Calculate total count and find min/max slots */
+	for (i = 0; i < slots_size; i++) {
+		if (slots[i] > 0) {
+			if (min_slot < 0)
+				min_slot = i;
+			max_slot = i;
+			total += slots[i];
+		}
+	}
+
+	if (min_slot < 0) {
+		*min = *max = *avg = *mean = *p95 = *p99 = 0;
```

Comment:
Instead of this, would it be better to explicitly return an appropriate failure code?